File: AquariumApi.DeviceApi/ClientApp/src/app/models/DeviceConfiguration.ts
import { Aquarium } from './Aquarium';
import { DeviceSchedule } from './DeviceSchedule';
import { DeviceScheduleTask } from './DeviceScheduleTask';
import { DeviceSensor } from './DeviceSensor';

export class DeviceConfiguration {
  aquarium: Aquarium
  mixingStation: object
  schedules: DeviceSchedule[]
  sensors: DeviceSensor[];
  tasks: DeviceScheduleTask[];
}

File: AquariumApi.Models/Account/DeviceConfiguration.cs
using System.Collections.Generic;

namespace AquariumApi.Models
{
    public class DeviceConfiguration
    {
        public Aquarium Aquarium { get; set; }

        public List<WirelessDevice> WirelessDevices { get; set; } = new List<WirelessDevice>();
        public List<DeviceSchedule> Schedules { get; set; } = new List<DeviceSchedule>();           //Schedules assigned to device
        public List<DeviceSensor> Sensors { get; set; } = new List<DeviceSensor>();                 //Sensors created on the device
        public List<DeviceScheduleTask> Tasks { get; set; } = new List<DeviceScheduleTask>();       //Tasks created on device
        public DeviceConfigurationSettings Settings { get; set; } = new DeviceConfigurationSettings();
        public DeviceConfigurationPrivateSettings PrivateSettings { get; set; } = new DeviceConfigurationPrivateSettings();
    }

}
